jpm: Incorrect libpath and headerpath

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

I was trying to use `jpm build`, which failed, see below.

According to `jpm show-paths`, jpm has two wrong paths:

```
libpath:    /nix/store/43bd68xzr2l6xjs09rigdjmkv8jqbgdz-jpm-1.1.0/lib
headerpath: /nix/store/43bd68xzr2l6xjs09rigdjmkv8jqbgdz-jpm-1.1.0/include/janet
```

The correct paths would be output paths of the janet package, like this:

```
libpath:    /nix/store/mqqv48inlhplym9ly5czwkg6s9zfjq8d-janet-1.37.143/lib
headerpath: /nix/store/mqqv48inlhplym9ly5czwkg6s9zfjq8d-janet-1.37.134/include/janet
```

### Steps to reproduce

With a nixpkgs-built jpm, try to build a binary:

```
$ jpm build
generating executable c source build/01.c from 01.janet...
compiling build/01.c to build/build___01.o...
linking build/01...
/nix/store/j7p46r8v9gcpbxx89pbqlh61zhd33gzv-binutils-2.43.1/bin/ld: cannot find /nix/store/43bd68xzr2l6xjs09rigdjmkv8jqbgdz-jpm-1.1.0/lib/libjanet.a: No such file or directory
collect2: error: ld returned 1 exit status
error: command failed with non-zero exit code 1
```

Workaround:

```
$ JANET_LIBPATH=/nix/store/mqqv48inlhplym9ly5czwkg6s9zfjq8d-janet-1.37.1/lib
 jpm build
```

### Expected behaviour

`jpm build` should work.

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.7, NixOS, 25.05 (Warbler), 25.05.20241229.88195a9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/8vz84mqgnm1gz5yk7hgnnb5gir5hjxas-source`

### Notify maintainers

@andrewchambers @peterhoeg

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/interpreters/janet/jpm.nix
+++ b/pkgs/development/interpreters/janet/jpm.nix
@@ -48,6 +48,10 @@ stdenv.mkDerivation rec {
 
     janet bootstrap.janet configs/${platformFile}
 
+    # patch default config to use janet's path instead of jpm itself
+    substituteInPlace $out/lib/janet/jpm/default-config.janet \
+      --replace-fail $out ${janet}
+
     runHook postInstall
   '';
 
@@ -55,6 +59,7 @@ stdenv.mkDerivation rec {
 
   installCheckPhase = ''
     $out/bin/jpm help
+    $out/bin/jpm show-paths
   '';
 
   meta = janet.meta // {

